java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -723,37 +854,25 @@ private void buildConditionalAnd(BinaryExpressionTree conditionalAnd, Block true
   private Block createBranch(Tree terminator, Block trueBranch, Block falseBranch) {
     Block result = createBlock();
     result.terminator = terminator;
-    result.successors.add(trueBranch);
-    result.successors.add(falseBranch);
+    result.addTrueSuccessor(trueBranch);
+    result.addFalseSuccessor(falseBranch);
     return result;
   }
 
   public void debugTo(PrintStream out) {
     for (Block block : Lists.reverse(blocks)) {
-      if (block.id != 0) {
-        out.println("B" + block.id + ":");
-      } else {
-        out.println("B" + block.id + " (Exit) :");
-      }
-      int i = 0;
-      for (Tree tree : block.elements()) {
-        out.println("  " + i + ": " + syntaxNodeToDebugString(tree));
-        i++;
-      }
-      if (block.terminator != null) {
-        out.println("  T: " + syntaxNodeToDebugString(block.terminator));
-      }
-      if (!block.successors.isEmpty()) {
-        out.print("  Successors:");
-        for (Block successor : block.successors) {
-          out.print(" B" + successor.id);
-        }
-        out.println();
-      }
+      block.debugTo(out);
     }
     out.println();
   }
 
+  @Override
+  public String toString() {
+    StringOutputStream buffer = new StringOutputStream();
+    debugTo(new PrintStream(buffer));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of constructor "PrintStream(OutputStream)" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1943)